updater.go

```diff
@@ -263,6 +334,62 @@ func (u *Updater) NeedUpdate(ctx Context) (upToDate bool, err error) {
 	return update.NeedUpdate, nil
 }
 
+func (u *Updater) CheckAndDownload(ctx Context) (availableAndDownloaded bool, err error) {
+	options := ctx.UpdateOptions()
+	update, err := u.checkForUpdate(ctx, options)
+
+	if err != nil {
+		return false, err
+	}
+
+	if !update.NeedUpdate {
+		return false, nil
+	}
+
+	// Linux updates don't have assets so it's ok to prompt for update above before
+	// we check for nil asset.
+	if update.Asset == nil || update.Asset.URL == "" {
```

Comment:
maybe this can be in a helper/method on `Update` since we use it in a few places now